src/crafting_gui.cpp

```diff
@@ -259,6 +259,13 @@ const recipe *select_crafting_recipe( int &batch_size )
                     return availability_cache[e];
                 } );
             }
+
+            // current/available have been rebuilt, make sure our cursor is still in range
+            if( current.empty() ) {
+                line = 0;
+            } else {
+                line = std::min( line, (int)current.size() - 1 );
```

Comment:
Are any of the lines broken at all? you need to delete the tabs. Does the file start and end with [ and ] ? 

Check every comma, quote, etc..